spec/open_api_spec.rb

```diff
@@ -89,10 +90,44 @@ def random_path
     end
   end
 
+  describe "Openapi schema attributes" do
+    PARSER_ATTRS = %w[readOnly nullable enum].freeze
+    context "correctly configured" do
+      ActiveRecord::Base.connection.tables.each do |table|
+        next if SKIP_TABLES.include?(table)
```

Comment:
The hard coded list is ok for now, but would be nice if we could find a more dynamic way to determine this.

Reply:
I looked into this. 
```
Api::Docs["1.0"].definitions || Api::Docs["1.0"].schemas
```
Doesn't return an array. The model needs to be known before hand before it will evaluate.
```
(byebug) doc.definitions
{}
(byebug) doc.definitions['blah']
*** ArgumentError Exception: Failed to find definition for blah

nil
(byebug) doc.definitions['OrderItem']
{"type"=>"object", "required"=>["service_parameters", "count", "service_plan_ref", "portfolio_item_id", "provider_control_parameters"], "properties"=>{"id"=>{"type"=>"string", "readOnly"=>true}, "count"=>{"type"=>"integer", "default"=>1, "title"=>"Item Count"}, "service_parameters"=>{"type"=>"object", "title"=>"JSON object with provisioning parameters"}, "provider_control_parameters"=>{"type"=>"object", "title"=>"Provider Control Parameters", "description"=>"The provider specific parameters needed to provision this service. This might include namespaces, special keys."}, "service_plan_ref"=>{"type"=>"string", "title"=>"The Service Plan ref", "description"=>"Stores the service plan ref from the Topology Service."}, "portfolio_item_id"=>{"type"=>"string", "title"=>"Portfolio Item ID", "description"=>"Stores the Portfolio Item ID."}, "state"=>{"type"=>"string", "enum"=>["Created", "Approval Pending", "Ordered", "Failed", "Completed", "Approved", "Denied"], "title"=>"State", "description"=>"Current state of this order item.", "readOnly"=>true}, "order_id"=>{"type"=>"string", "title"=>"Order ID", "description"=>"The Order that the OrderItem belongs to."}, "created_at"=>{"type"=>"string", "format"=>"date-time", "title"=>"Created at", "readOnly"=>true}, "order_request_sent_at"=>{"type"=>"string", "format"=>"date-time", "title"=>"Order request sent at", "description"=>"The time at which the order request was sent to the Topology Service", "readOnly"=>true}, "completed_at"=>{"type"=>"string", "format"=>"date-time", "title"=>"Completed at", "readOnly"=>true}, "updated_at"=>{"type"=>"string", "format"=>"date-time", "title"=>"Completed at", "readOnly"=>true}, "owner"=>{"type"=>"string", "title"=>"Owner", "example"=>"jdoe", "readOnly"=>true}, "external_url"=>{"type"=>"string", "title"=>"External url", "description"=>"The external url of the service instance used with relation to this order item"}, "insights_request_id"=>{"type"=>"string", "title"=>"Insights Request ID ", "example"=>"364498f142194beba576833d7303abe5", "description"=>"The insights request id can be used to collect log data for this order item as its processed by the system", "readOnly"=>true}}}
(byebug) 
```
So we need to pre seed our AR model list.